client/blocks/site-icon/docs/example.jsx

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import SiteIcon from '../';
+import { getCurrentUser } from 'state/current-user/selectors';
+
+/**
+ * Site ID of en.blog.wordpress.com, to be used as fallback for SiteIcon if
+ * current user does not have a primary site.
+ *
+ * @type {Number}
+ */
+const EN_BLOG_SITE_ID = 3584907;
+
+const SiteIconExample = ( { siteId } ) => <SiteIcon siteId={ siteId || EN_BLOG_SITE_ID } />;
```

Comment:
An alternative approach might be to forgo the intermediate `SiteIconExample` component and simply wrap the raw `SiteIcon` component with `connect` directly. Doing so would require moving the fallback into the `connect` selector, but it would be a good example of the underlying component being agnostic to where the incoming props originate / are determined.